omegaconf/_impl.py

```diff
@@ -0,0 +1,44 @@
+from typing import Any
+
+from omegaconf import MISSING, Container, DictConfig, ListConfig, Node, ValueNode
+from omegaconf.errors import InterpolationToMissingValueError
+
+
+def _resolve_container_value(cfg: Container, node: Node, index: Any) -> None:
+    if node._is_interpolation():
+        try:
+            resolved = node._dereference_node()
+            assert resolved is not None
+            if isinstance(resolved, Container) and isinstance(node, ValueNode):
+                cfg[index] = resolved
+            else:
+                node._set_value(resolved._value())
+        except InterpolationToMissingValueError:
+            node._set_value(MISSING)
+    else:
+        _resolve(node)
+
+
+def _resolve(cfg: Node) -> Node:
+    assert isinstance(cfg, Node)
+    try:
+        if cfg._is_interpolation():
+            resolved = cfg._dereference_node()
+            assert resolved is not None
+            cfg._set_value(resolved._value())
+    except InterpolationToMissingValueError:
+        cfg._set_value(MISSING)
```

Comment:
MInor (makes it clearer where the exception may be raised):
```python
    if cfg._is_interpolation():
        try:
            resolved = cfg._dereference_node()
        except InterpolationToMissingValueError:
            cfg._set_value(MISSING)
        else:
            assert resolved is not None
            cfg._set_value(resolved._value())
```

Reply:
I don't like this syntax very much, feels like it's breaking the flow too much.
trying it to see if it grows on me.

Reply:
I agree it's not the prettiest, but after you spend a lot of time investigating a tricky bug caused by a too large `try...except` block, it grows on you pretty fast ;)